### PR TITLE
Feat/13 run only one test on a branch at a time

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
# Update GitHub Actions Workflow

## Changes

- **Concurrency**: Ensure only one run per branch at a time.

- **Solution**: https://github.com/orgs/community/discussions/26940

@lasuk Please review the changes and provide feedback or approval for merging.

### CODE REVIEW

File: .github/workflows/check-release.yml [hunk 1]
```diff
@@ -17,13 +16,18 @@ jobs:
...
-    # by "dependabot[bot]". Otherwise, clone it normally.
       CC_API_ORGANISATION: ${{ secrets.CC_API_ORGANISATION }}
       CC_API_KEY: ${{ secrets.CC_API_KEY }}
-    if: |
-        (github.event_name == 'pull_request_target' && github.actor == 'dependabot[bot]') ||
-        (github.event_name != 'pull_request_target' && github.actor != 'dependabot[bot]')
+    if: >
+      (github.event_name != 'pull_request' && !github.event.pull_request.head.repo.fork) ||
+      (github.event_name == 'pull_request' &&
+      (github.event.pull_request.head.repo.fork || github.actor == 'dependabot[bot]'))
```
Comment (lasuk) [inline .github/workflows/check-release.yml:24]: These lines are cryptic. Best to leave a comment telling us what theses lines good for / why we need this snippet.

Note: This file fiddles with configurations and is rather hard to understand for an unexperienced user (like me). Contrary to our usual Python code where we target code readability, we need comments here for clarity.

File: .github/workflows/check-release.yml [hunk 2]
```diff
@@ -17,13 +16,18 @@ jobs:
...
+
+    # Add concurrency to ensure only one run at a time per branch
+    concurrency:
+      group: ${{ github.ref }}
```
Comment (lasuk) [inline .github/workflows/check-release.yml:28]: What does '${{ github.ref }}' stand for? What is the effect of using it?

**Review by lasuk — COMMENTED** [review]
Thanks for fixing this issue. I've added a few questions to better understand what is going on in GitHub action configurations.

File: .github/workflows/check-release.yml (hunk 1 above)
Comment (AlexTheWizardL) [reply .github/workflows/check-release.yml:32]: Added

File: .github/workflows/check-release.yml (hunk 2 above)
Comment (AlexTheWizardL) [reply .github/workflows/check-release.yml]: Actually, this lines is redundant for our case, dropped

**Comment by lasuk** [comment]
Feature #13 was Implemented in #22 yesterday.

**Pull request closed without merging**